lib/modules/manager/bazel-module/extract.ts

```diff
@@ -0,0 +1,19 @@
+import type { PackageFileContent } from '../types';
+import * as bazelDep from './bazel-dep';
+import { isNotNullOrUndefined } from './filters';
+import { parse } from './parser';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string
+): PackageFileContent | null {
+  const fragments = parse(content, packageFile);
+  if (!fragments) {
+    return null;
+  }
+  const deps = fragments
+    .map((frag) => bazelDep.toPackageDependency(frag))
+    .filter(isNotNullOrUndefined);
+  // istanbul ignore next: cannot reach null without introducing fake rule
```

Comment:
can null be reached at runtime? if not, just return the object and skip the empty check.

Reply:
I don't follow your meaning. The `toPackageDependency()` function can return `null` values. Hence, I added the filter to ensure that `deps` only contains `PackageDependency` instances.

Reply:
can't you create a test case where the length of deps will be zero? can it happen on real repos?